cmd/kube-scheduler/app/options/insecure_serving.go

```diff
@@ -31,8 +31,8 @@ import (
 // CombinedInsecureServingOptions sets up to two insecure listeners for healthz and metrics. The flags
 // override the ComponentConfig and DeprecatedInsecureServingOptions values for both.
 type CombinedInsecureServingOptions struct {
-	Healthz *apiserveroptions.DeprecatedInsecureServingOptions
-	Metrics *apiserveroptions.DeprecatedInsecureServingOptions
+	Healthz *apiserveroptions.DeprecatedInsecureServingOptionsWithLoopback
+	Metrics *apiserveroptions.DeprecatedInsecureServingOptionsWithLoopback
```

Comment:
it seems that is it extending it to loopback? though still not sure why to extend a deprecated option?

Reply:
It's not extending it in the code sense. We have support for insecure ports using these two option structs (not new for scheduler). The whole insecure port code is deprecated, to be removed when all components switched over to secure ports. The difference of `DeprecatedInsecureServingOptionsWithLoopback` is just that it add the loopback authenticator which helps testing. Otherwise it is the same.